tox.ini

```diff
@@ -0,0 +1,35 @@
+[tox]
+requires = tox>=4
+env_list = py{37,38,39,310,311,312}, lint, mypy, black
+
+[testenv]
+description = Run unit tests
+package = wheel
+extras = test
+pass_env = SSH_*
+commands = pytest --color=yes {posargs}
+
+[testenv:lint]
+description = Lint via pre-commit
+base_python = py39
+commands = pre-commit run --all-files
+
+[testenv:mypy]
+description = Typecheck with mypy
+base_python = py39
+commands = mypy -p git
+ignore_outcome = true
+
+[testenv:black]
+description = Check style with black
```

Comment:
If black is the most prominent formatter by now one could consider to reformat the entire codebase to match. That might make contributions easier as it avoids spurious changes when saving while an auto-formatting is enabled.

Reply:
Good idea. I have done this in #1668.